index/postings_test.go

```diff
@@ -532,3 +532,14 @@ func TestIntersectWithMerge(t *testing.T) {
 	testutil.Ok(t, err)
 	testutil.Equals(t, []uint64{30}, res)
 }
+
+func TestReduceToEmptyPostings(t *testing.T) {
+	a := newListPostings([]uint64{1, 2, 3})
+	b := newListPostings([]uint64{2, 3, 4})
+	c := EmptyPostings()
+
+	testutil.Assert(t, Intersect(a, b, c) == EmptyPostings(), "was not empty postings")
```

Comment:
the test fails when the empty posting is not the last one in the list
`testutil.Assert(t, Intersect(c, a, b) == EmptyPostings(), "was not empty postings")`

Reply:
ping @gouthamve , @bwplotka

Reply:
hmm

cannot repro it. This below is fine:

```
func TestReduceToEmptyPostings(t *testing.T) {
	a := newListPostings([]uint64{1, 2, 3})
	b := newListPostings([]uint64{2, 3, 4})
	c := EmptyPostings()

	testutil.Assert(t, Intersect(a, b, c) == EmptyPostings(), "was not empty postings")
	testutil.Assert(t, Intersect(c, b, a) == EmptyPostings(), "was not empty postings")
	testutil.Assert(t, Intersect(a, c, b) == EmptyPostings(), "was not empty postings")
	testutil.Assert(t, Intersect(c, a, b) == EmptyPostings(), "was not empty postings")
}
```

Reply:
Anyway, rebasing (merge was changed a lot) and testing more

Reply:
yeah maybe the bug was somewhere else and it was fixed at some point.